flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -5176,18 +5252,11 @@ class ArrayExprLowering {
       // always loaded at the beginning of the statement and merged at the
       // end.
       destination = arrLoad;
-      return [=](IterSpace iters) -> ExtValue {
-        auto innerArg = iters.innerArgument();
-        auto resTy = innerArg.getType();
-        auto eleTy = fir::applyPathToType(resTy, iters.iterVec());
-        auto element = tryUpdateConversions(eleTy, iters.getElement());
-        auto arrUpdate = builder.create<fir::ArrayUpdateOp>(
-            loc, resTy, innerArg, element, iters.iterVec(),
-            destination.typeparams());
-        return abstractArrayExtValue(arrUpdate);
+      return [lambda = ccStoreToDest](IterSpace iters) -> ExtValue {
```

Comment:
Isn't `return ccStoreToDest;` equivalent ?

Reply:
We want to pass the closure that is in `this->ccStoreToDest` at this point of the evaluation and not whatever `this->ccStoreToDest` closure will be when the continuation is evaluated. This forces a copy.

Reply:
No longer necessary.